manifests/share.pp

```diff
@@ -52,11 +52,17 @@
   }
 
   if $ensure == present {
+    if $comment == undef {
+        fail('parameter "comment" must be set')
+    }
+    if $path == undef {
+        fail('parameter "path" must be set')
+    }
 
     samba::option {
       "${title}-comment":              target => $target, key => 'comment',  value => $comment;
       "${title}-path":                 target => $target, key => 'path',     value => $path;
-      "${title}-writable":             target => $target, key => 'writable', value => $writable;
+      "${title}-writeable":            target => $target, key => 'writeable', value => $writeable;
```

Comment:
The SAMBA documentation uses `writable`. Please see here:

https://www.samba.org/samba/docs/using_samba/ch09.html

Reply:
Thanks for the PR, as English is not my first language, I was unsure, what are the correct spelling for `writable`. However, I looks like the SAMBA docs use it as `writable` and not `writeable`.

https://www.samba.org/samba/docs/using_samba/ch09.html

Also please squash your commits into one.

Reply:
> The SAMBA documentation uses `writable`. Please see here:
> 
> https://www.samba.org/samba/docs/using_samba/ch09.html

That is old documentation.  If you look [here](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html#WRITEABLE) you can see it the other way.  Both forms are accepted but as [this](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html#WRITABLE) notes, `writable` is a synonym for `writeable`.  That would lead me to believe the longer one is what they consider conventional while the shorter one is an alternative but also accepted.  However, en.wiktionary.org seems to state it the other way around.  English *is* my first language and I can't find anything conclusive as to which is correct.  The short spelling struck me as just plain wrong at first look, however.  Aliases are everywhere in other tech, but I'm not sure the same is possible in Puppet.

So I don't want to squash anything until you decided whether to keep the original `writable` spelling or use my commit to change the spelling.  Why squash anyway?  That takes away the ability to drop individual commits among other things.

Reply:
If the `writable` works as well and is not a bug then I would like to leave it as is. I asked my colleagues at work and people mostly spelled `writable`.
Also, I want to keep the other changes as one logical commit and keep the commit history clean. Some please rebase it and remove the changes for `writeable`. 

Thanks.

Reply:
Fair enough.